go/client/client.go

```diff
@@ -61,12 +61,12 @@ type Client struct {
 	// Retrier is the Retrier that is used for RPCs made by this client.
 	//
 	// This field is logically "protected" and is intended for use by extensions of Client.
-	Retrier        *Retrier
-	chunkMaxSize   ChunkMaxSize
-	useBatchOps    UseBatchOps
-	casConcurrency CASConcurrency
-	rpcTimeout     time.Duration
-	creds          credentials.PerRPCCredentials
+	Retrier      *Retrier
+	chunkMaxSize ChunkMaxSize
+	useBatchOps  UseBatchOps
+	casUploaders chan bool
```

Comment:
I think semaphores are a more natural fit here.

Reply:
Not according to this: http://jmoiron.net/blog/limiting-concurrency-in-go/ which was the second result on controlling concurrency in Go. The first result (https://blog.golang.org/pipelines) agrees with this pattern as well. In general, the guideline is "you should rarely need to use sync package primitives, use channels instead".

Reply:
Discussed offline. sgtm.